docs/csharp/language-reference/compiler-options/refonly-compiler-option.md

```diff
@@ -0,0 +1,47 @@
+---
+title: "-refonly (C# Compiler Options)"
+ms.date: "2017-07-08"
+ms.prod: .net
+ms.technology: 
+  - "devlang-csharp"
+ms.topic: "article"
+f1_keywords: 
+  - "/refonly"
+dev_langs: 
+  - "CSharp"
+helpviewer_keywords: 
+  - "/refonly compiler option [C#]"
+  - "-refonly compiler option [C#]"
+author: "BillWagner"
+ms.author: "wiwagn"
+---
+
+# /refonly (C# Compiler Options)
+
+The **/refonly** The /refonly option indicates that a reference assembly should be output instead of an implementation assembly, as the primary output. The `/refonly` parameter silently disables outputting PDBs, as reference assemblies cannot be executed.
+
+## Syntax
+
+```console
+/refonly
+```
+
+## Remarks
+
+Metadata-only assemblies have their method bodies replaced with a single `throw null` body, but include all members except anonymous types. The reason for using `throw null` bodies (as opposed to no bodies) is so that PEVerify could run and pass (thus validating the completeness of the metadata).
+
+Reference assemblies include an assembly-level `ReferenceAssembly` attribute. This attribute may be specified in source (then the compiler won't need to synthesize it). Because of this attribute, runtimes will refuse to load reference assemblies for execution (but they can still be loaded in reflection-only mode). Tools that reflect on assemblies need to ensure they load reference assemblies as reflection-only, otherwise they will receive a typeload error from the runtime.
+
+Reference assemblies further remove metadata (private members) from metadata-only assemblies:
+
+- A reference assembly only has references for what it needs in the API surface. The real assembly may have additional references related to specific implementations. For instance, the reference assembly for `class C { private void M() { dynamic d = 1; ... } }` does not reference any types required for `dynamic`.
+- Private function-members (methods, properties, and events) are removed. If there are no `InternalsVisibleTo` attributes, do the same for internal function-members.
```

Comment:
This is not necessarily true. There are still a few number of cases where private members are maintained because the observably impact compilation. It's hard to enumerate the scenarios here. May be easier to say they are removed in cases where it doesn't observably impact compilation.